### PR TITLE
Tv reducer test

### PR DESCRIPTION
**Pull request by alisherk**

### CODE REVIEW

File: front/src/store/tv/__tests__/reducer.test.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  it('saves episode data', () => {
+    const expectedState = {
+      savedQueries: [],
+      episodeData: { episode: 'test' },
+      basicShowInfo: {},
+      calendarEpisodesForDisplay: [],
+    };
+    const action = { type: SAVE_EPISODE_DATA, payload: { episode: 'test' } };
+    expect(tvReducer(undefined, action)).toEqual(expectedState);
```
Comment (trybick) [inline front/src/store/tv/__tests__/reducer.test.ts:34]: should this `undefined` be `initialState` instead?

File: front/src/store/tv/__tests__/reducer.test.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    calendarEpisodesForDisplay: [],
+  };
+
+  it('saves query', () => {
+    const action = { type: SAVE_SEARCH_QUERY, payload: { query: '1' } };
+    expect(tvReducer(undefined, action)).toEqual(commonState);
+  });
+
+  it('does not add same query', () => {
+    const initialState = {
```
Comment (trybick) [inline front/src/store/tv/__tests__/reducer.test.ts]: this `initialState` looks the same as commonState above. do we need it?

File: front/src/store/tv/__tests__/reducer.test.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      savedQueries: [],
+      episodeData: {},
+      basicShowInfo: { 0: { episode: 'test' } },
+      calendarEpisodesForDisplay: [],
+    };
+    const action = { type: REQUEST_BASIC_SHOW_INFO_SUCCEEDED, payload: [{ episode: 'test' }] };
+    expect(tvReducer(undefined, action)).toEqual(expectedState);
+  });
+
+  it('returns calendar episode for display', () => {
```
Comment (trybick) [inline front/src/store/tv/__tests__/reducer.test.ts]: we should be consistent with the naming of the tests - so the last two say "returns" and the first few same "saves". it might be better for all the test names to follow a pattern

File: front/src/store/tv/__tests__/reducer.test.ts (hunk 2 above)
Comment (trybick) [reply front/src/store/tv/__tests__/reducer.test.ts]: Also I'm not sure if we need to test that it doesn't duplicate the query, so maybe we can remove this test. But let me know if you think we should keep it

File: front/src/store/tv/__tests__/reducer.test.ts (hunk 1 above)
Comment (alisherk) [reply front/src/store/tv/__tests__/reducer.test.ts:34]: It can be but we don't need to define it except in the 'does not add same query' test

File: front/src/store/tv/__tests__/reducer.test.ts (hunk 2 above)
Comment (alisherk) [reply front/src/store/tv/__tests__/reducer.test.ts]: It is the core logic of the reducer on this action. So I believe we should but I can remove it as well.  initialState is required here because we need to have savedQuery array to contain something to be able to test filtering.

File: front/src/store/tv/__tests__/reducer.test.ts (hunk 3 above)
Comment (alisherk) [reply front/src/store/tv/__tests__/reducer.test.ts]: Okay I will look for a more consistent pattern.

File: front/src/store/tv/__tests__/reducer.test.ts (hunk 1 above)
Comment (trybick) [reply front/src/store/tv/__tests__/reducer.test.ts:34]: Ok, keeping as you have it looks good

File: front/src/store/tv/__tests__/reducer.test.ts (hunk 2 above)
Comment (trybick) [reply front/src/store/tv/__tests__/reducer.test.ts]: Ok sounds good keeping it. Except `initialState` is the same as `commonState` above. Can we use commonState?

File: front/src/store/tv/__tests__/reducer.test.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  });
+
+  it('returns expected state on SAVE_EPISODE_DATA', () => {
+    const expectedState = {
+      savedQueries: [],
+      episodeData: { episode: 'test' },
+      basicShowInfo: {},
+      popularShows: [],
+      calendarEpisodesForDisplay: [],
+    };
```
Comment (trybick) [inline front/src/store/tv/__tests__/reducer.test.ts]: I think we can avoid writing the whole state in each block by renaming commonState at the top to blankState and then in each test, like here, do:

```
const expectedState = { ...blankState, episodeData: { episode: 'test' } }
```

Comment (alisherk) [reply front/src/store/tv/__tests__/reducer.test.ts]: @trybick  A great suggestion!

**Comment by trybick** [comment]
thanks @alisherk !
